tsconfig.json

```diff
@@ -1,3 +1,3 @@
 {
-  "extends": "./node_modules/gts/tsconfig-google.json"
+  "extends": "./tsconfig.base.json"
```

Comment:
What is the logic for tsconfig.json for all packages ?

In some of the changes, I see you have modified some package to no longer extend from base. For example, search-token-server or search-token-lambda tsconfig is quite heavily modified.

Also, why do we have two files at top level `tsconfig.base.json` and `tsconfig.json` ?

It seems `tsconfig.json` is not imported/extended by any package. What is it's utility ? Is it the compile config from something ?

Reply:
>Also, why do we have two files at top level tsconfig.base.json and tsconfig.json ?

The reasons for this change is explained here:
https://github.com/coveo/cli/pull/632#discussion_r794503735

> It seems tsconfig.json is not imported/extended by any package. What is it's utility ? Is it the compile config from something ?
Just typing crutches for the other js/ts files outside of a package, could be removed tho I reckon

> In some of the changes, I see you have modified some package to no longer extend from base. For example, search-token-server or search-token-lambda tsconfig is quite heavily modified.

They were extending `gts`, which become impossible/painful with the npm workspace, because there's only one node_module folder and its at the root.

MEaning that I would have to do `../../node_modules/gts/tsconfig.json`. But the tsconfig.json are packaged, and that path would refer to outside the package, so that was a no go